### PR TITLE
Adding changes to support Private DNS Public Preview

### PR DESCRIPTION
**Pull request by kay-owolabi**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

PR Based on https://github.com/Azure/azure-rest-api-specs/pull/2558

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Review by muwaqar — APPROVED** [review]

**Comment by sarangan12** [comment]
1. https://travis-ci.org/Azure/azure-rest-api-specs/jobs/348509126
2. https://travis-ci.org/Azure/azure-rest-api-specs/jobs/348509127

Please take a look at the list of the errors in the above links. The errors must be fixed for approval

File: specification/dns/resource-manager/Microsoft.Network/preview/2018-03-01-preview/dns.json [hunk 1]
```diff
@@ -1442,8 +1442,38 @@
...
+        "resolutionVirtualNetworks": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SubResource"
+          },
+          "description": "A list of references to virtual networks that resolve records in this DNS zone. This is a only when ZoneType is Private."
         }
       },
+      "required": [
+        "zoneType"
```
Comment (muwaqar) [inline specification/dns/resource-manager/Microsoft.Network/preview/2018-03-01-preview/dns.json]: Is having `required` here a breaking change? Cannot we default it to `Public`?

**Review by muwaqar — COMMENTED** [review]
Is having `zoneType` as `required` a breaking change? Cannot we default it to Public?

**Comment by muwaqar** [comment]
Looks like `zoneType` already has a default value of `Public` (from line 1456).

**Comment by salameer** [comment]
@sarangan12 anymore feedback from your Side?

**Comment by sarangan12** [comment]
@owolabi2 One of the jobs is failing - https://travis-ci.org/Azure/azure-rest-api-specs/jobs/350107652#L7099. This is not related to your PR directly. Let me explain:

In your branch, Look at the line: https://github.com/owolabi2/azure-rest-api-specs/blob/a562583fa70c1ba174dd41aa21686e1eac81ff5b/specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2017-03-01/apimdiagnostics.json#L541 

Now, Look at the same line in the master branch: https://github.com/Azure/azure-rest-api-specs/blob/master/specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2017-03-01/apimdiagnostics.json#L541

Please merge the changes from master branch. This will help the job to proceed further and can point to any errors (if available) in your PR.

Now, apart from the above error, Please take a look at the errors reported in this job: https://travis-ci.org/Azure/azure-rest-api-specs/jobs/350107662 These errors are related to this PR and must be fixed. If you need any help in resolving these errors, Please let me know.

**Comment by ravbhatnagar** [comment]
Signing off!

**Review by sarangan12 — APPROVED** [review]

**Pull request merged**